src/chain_access.rs

```diff
@@ -0,0 +1,190 @@
+use crate::errors::ChainSyncError;
+use bitcoin::{Script, Transaction, Txid};
+use esplora_client::AsyncClient;
+use lightning::chain::{Confirm, Filter, WatchedOutput};
+use std::sync::{Arc, Mutex};
+
+#[allow(dead_code)]
+pub(crate) struct LipaChainAccess {
+    esplora_client: Arc<AsyncClient>,
+    queued_txs: Mutex<Vec<(Txid, Script)>>,
+    watched_txs: Mutex<Vec<(Txid, Script)>>,
+    queued_outputs: Mutex<Vec<WatchedOutput>>,
+    watched_outputs: Mutex<Vec<WatchedOutput>>,
+}
+
+impl LipaChainAccess {
+    pub(crate) fn new(esplora_client: Arc<AsyncClient>) -> Self {
+        let queued_txs = Mutex::new(Vec::new());
+        let watched_txs = Mutex::new(Vec::new());
+        let queued_outputs = Mutex::new(Vec::new());
+        let watched_outputs = Mutex::new(Vec::new());
+        Self {
+            esplora_client,
+            queued_txs,
+            watched_txs,
+            queued_outputs,
+            watched_outputs,
+        }
+    }
+
+    #[allow(dead_code)]
+    pub(crate) fn sync(
+        &self,
+        _confirmables: Vec<&(dyn Confirm + Sync)>,
+    ) -> Result<(), ChainSyncError> {
+        // TODO: sync with the chain
+        // this will include moving queued txs and outputs to watched ones
+
+        Ok(())
+    }
+}
+
+impl Filter for LipaChainAccess {
+    fn register_tx(&self, txid: &Txid, script_pubkey: &Script) {
+        self.queued_txs
+            .lock()
+            .unwrap()
+            .push((*txid, script_pubkey.clone()));
+    }
+
+    fn register_output(&self, output: WatchedOutput) -> Option<(usize, Transaction)> {
+        self.queued_outputs.lock().unwrap().push(output);
+        // The Filter::register_output return value has been removed, as it was very difficult to
+        // correctly implement (i.e., without blocking). Users previously using it should instead
+        // pass dependent transactions in via additional chain::Confirm::transactions_confirmed
+        // calls (#1663).
+        None
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::LipaChainAccess;
+    use bitcoin::consensus::deserialize;
+    use bitcoin::hashes::hex::FromHex;
+    use bitcoin::Transaction;
+    use esplora_client::Builder;
+    use lightning::chain::transaction::OutPoint;
+    use lightning::chain::{Filter, WatchedOutput};
+    use std::sync::Arc;
+
+    #[test]
+    fn filter_is_initialised_empty() {
+        // 9 is a discard port
+        // See https://en.wikipedia.org/wiki/Port_(computer_networking)
+        let builder = Builder::new("http://localhost:9");
+        let esplora_client = Arc::new(builder.build_async().unwrap());
+
+        let filter = LipaChainAccess::new(esplora_client.clone());
```

Comment:
Create a `fn setup()` within the `tests` mod for the code up until here.
This will duplicated code and particularly prevent us from having to read that port 9 is a discard port over and over again